upup/pkg/fi/cloudup/awstasks/launchtemplate_target_terraform.go

```diff
@@ -212,11 +210,10 @@ func (t *LaunchTemplate) RenderTerraform(target *terraform.TerraformTarget, a, e
 		if err != nil {
 			return err
 		}
-		b64d := base64.StdEncoding.EncodeToString(d)
-		if b64d != "" {
-			b64UserDataResource := fi.WrapResource(fi.NewStringResource(b64d))
+		if d != nil {
+			userDataResource := fi.WrapResource(fi.NewBytesResource(d))
```

Comment:
I suppose i could check the value of the terraform feature flag here and conditionally encode the userdata contents, but that feels ugly. any other ideas?

Reply:
or add a new `target.AddBase64File` function that wraps `target.AddFile`

Reply:
Seems like it's in the nature of the feature flag to be ugly. A nice reminder that has to go away soon. 😄
I would go with the flag.

Reply:
done 👍  `hack/verify-terraform.sh` passes too. I'm not sure why there isnt a prow job for it, i'll look into creating one.

Reply:
LGTM, but still WIP and also the new job will need some work to pass.